core/ustring.cpp

```diff
@@ -96,37 +96,95 @@ bool select_word(const String &p_s, int p_col, int &r_beg, int &r_end) {
 	}
 }
 
-/** STRING **/
+/*************************************************************************/
+/*  Char16String                                                         */
+/*************************************************************************/
 
-bool CharString::operator<(const CharString &p_right) const {
+bool Char16String::operator<(const Char16String &p_right) const {
 	if (length() == 0) {
 		return p_right.length() != 0;
 	}
 
 	return is_str_less(get_data(), p_right.get_data());
 }
 
-CharString &CharString::operator+=(char p_char) {
+Char16String &Char16String::operator+=(char16_t p_char) {
 	resize(size() ? size() + 1 : 2);
 	set(length(), 0);
 	set(length() - 1, p_char);
 
 	return *this;
 }
 
-const char *CharString::get_data() const {
+Char16String &Char16String::operator=(const char16_t *p_cstr) {
+	copy_from(p_cstr);
+	return *this;
+}
+
+const char16_t *Char16String::get_data() const {
 	if (size()) {
 		return &operator[](0);
 	} else {
-		return "";
+		return u"";
+	}
+}
+
+void Char16String::copy_from(const char16_t *p_cstr) {
+	if (!p_cstr) {
+		resize(0);
+		return;
+	}
+
+	const char16_t *s = p_cstr;
+	for (; *s; s++) {
+	};
```

Comment:
Seems to be an unnecessary semicolon?

Reply:
Yep, if was `for (; *s; s++);` and clang-format added `{}`.

Reply:
Ah yeah I got that too when formatting. Then I usually change it to `for (; *s; s++) {}` which I think `clang-format` might be happy with.
But it's no big deal and doesn't warrant an amend just for this anyway.